zookeeper-server/src/main/java/org/apache/zookeeper/OpResult.java

```diff
@@ -170,6 +173,67 @@ public int hashCode() {
         }
     }
 
+    /**
+     * A result from a getChildren operation. Provides a list which contains
+     * the names of the children of a given node.
+     */
+    public static class GetChildrenResult extends OpResult {
+        private List<String> children;
+
+        public GetChildrenResult(List<String> children) {
+            super(ZooDefs.OpCode.getChildren);
+            this.children = children;
+        }
+
+        public List<String> getChildren() {
+            return children;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (!(o instanceof GetChildrenResult)) return false;
+
+            GetChildrenResult other = (GetChildrenResult) o;
+            return getType() == other.getType() && children.equals(other.children);
+        }
+
+        @Override
+        public int hashCode() {
+            return getType() * 35 + children.hashCode();
+        }
+    }
+
+    /**
+     * A result from a getData operation. The data is represented as a byte array.
+     */
+    public static class GetDataResult extends OpResult {
+        private byte[] data;
+
+        public GetDataResult(byte[] data) {
+            super(ZooDefs.OpCode.getData);
+            this.data = data.clone();
```

Comment:
Consider Arrays.copyOf() instead of clone()?  clone feels antiquated to me, but I'm not clear what the best practice is...

Check for null on data?

this.data = (data == null ? null : Arrays.copyOf(data, data.length));